apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,34 +1,51 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
+import { NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChild,
+  EventEmitter,
+  Input,
+  Output,
+} from '@angular/core';
 import { CardType } from '../../model/card.model';
 import { ListItemComponent } from '../list-item/list-item.component';
+import { CardTemplateContentDirective } from './card-template-content.directive';
 
 @Component({
   selector: 'app-card',
   templateUrl: './card.component.html',
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [
+    NgIf,
+    NgFor,
+    ListItemComponent,
+    NgTemplateOutlet,
+    CardTemplateContentDirective,
+  ],
+  styles: [
+    `
+      :host > div {
+        background-color: var(--card-bg-color);
+      }
+    `,
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class CardComponent {
-  @Input() list: any[] | null = null;
+export class CardComponent<T> {
+  @Input() list: T[] | null = null;
   @Input() type!: CardType;
   @Input() customClass = '';
 
+  @Output() addItemEvent: EventEmitter<void> = new EventEmitter<void>();
+
   CardType = CardType;
 
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore
-  ) {}
+  @ContentChild(CardTemplateContentDirective)
+  template!: CardTemplateContentDirective;
+
+  constructor() {}
```

Comment:
nit: you should always delete useless code. It will add some extra bytes in your final bundle